js/apps/admin-ui/src/organizations/LinkIdentityProviderModal.tsx

```diff
@@ -77,6 +89,13 @@ export const LinkIdentityProviderModal = ({
         ...config,
       };
       foundIdentityProvider.hideOnLogin = data.hideOnLogin;
+      if (emailMatchAnyOrgDomain === "true") {
+        delete foundIdentityProvider.config[
+          "kc.org.broker.redirect.mode.email-matches"
+        ];
+        delete foundIdentityProvider.config["kc.org.domain"];
+      }
+
```

Comment:
I'm not a fan of this, but I don't see an easy way to get around this

Reply:
No idea myself, I mostly work on the backend side :)